spread/lxd.go

```diff
@@ -111,6 +127,16 @@ func (p *lxdProvider) Allocate(ctx context.Context, system *System) (Server, err
 	if !p.options.Reuse {
 		args = append(args, "--ephemeral")
 	}
+	if p.vm {
+		args = append(args, "--vm")
+	}
+	if p.backend.Memory > 0 {
+		mem := int(p.backend.Memory / mb)
+		args = append(args, "-c", fmt.Sprintf("limits.memory=%dMiB", mem))
+	}
+	if system.Storage != Size(0) {
+		args = append(args, "-d", fmt.Sprintf("root,size=%d", system.Storage))
+	}
```

Comment:
I see no way to specify the number of vCPUs for the VM. This is an important requirement as the default of 1 vCPU is most likely not sufficient.

In https://github.com/canonical/spread/pull/221/commits/804d1ee7fce005ba0c275bc24adc3318fe61ef7c generic support for the LXD instance type concept is introduced driven by the already existing `plan` keyword.

LXD applies these constraints both to containers and VMs.